virtual_rainforest/models/animals/animal_communities.py

```diff
@@ -135,7 +147,7 @@ def birth_community(self) -> None:
         """This handles birth for all cohorts in a community."""
 
         # Create a snapshot list of the current cohorts
-        current_cohorts = list(chain.from_iterable(self.animal_cohorts.values()))
+        current_cohorts = list(self.all_animal_cohorts)
 
         # reproduction occurs for cohorts with sufficient energy
         for cohort in current_cohorts:
```

Comment:
Unless I'm missing something, this can just be:

```suggestion
        # reproduction occurs for cohorts with sufficient energy
        for cohort in self.all_animal_cohorts:
```

Reply:
The reason I had done it this way is that the first pass just called a birth process with no energetic gatekeeping, which caused newly born cohorts to also give birth and so on. I'll change it now.

Reply:
Ah I see. Makes sense.